_posts/2024-02-17-survey-https-infra.md

```diff
@@ -0,0 +1,194 @@
+
+## BaNaNaS bandwidth bill
+
+Hooking in to the above, the move to HTTPS also has another goal: about 10% of our bandwidth for BaNaNaS is till this day done via TCP.
+Although the HTTP(S) downloads are free of charge (thanks to [Cloudflare](https://www.cloudflare.com/)), the TCP downloads are not.
+
+To be more exact, as they are served by AWS, it costs 0.90 dollar per GB transferred.
+And this grows to a large number very quickly; over 50% of our monthly bill is only for BaNaNaS downloads.
+
+Now there are many ways to mitigate this, ranging from "migrate away from AWS" to "just disable TCP downloads".
+But for now, I have good hope that the change to HTTPS will move even more traffic away from TCP.
+Related to that, we also removed the setting that forced usage of TCP over HTTPS: OpenTTD now always first tries HTTPS, and if that fails, it falls back to TCP.
+Hopefully that sufficiently reduces the bandwidth bill to the bare minimum.
+
+If not, we will need to take action on this.
+I keep hoping Cloudflare releases the ability to route TCP (and not just HTTP(S)) via their infrastructure; they keep teasing this in their blog-posts, but they have since 2021.
+So at a certain point I have to give up that hope, and find a better solution myself.
+
+## Infrastructure
+
+[Our migration](https://www.openttd.org/news/2023/07/09/infra-migration) in 2023 was a great success, and is till this day.
+Deployments have never been easier, and maintenance has been a breeze.
+Auto-healing and incident resolving work as expected, and in general, [Nomad](https://www.nomadproject.io/) has been a good fit.
+This combined with our new GitHub workflows, there is a clear noticeable difference in how much time is required to keep everything running (for the better, ofc).
+
+Our Cloudflare bill has been very stable; which is very good for a project like OpenTTD, as predictability means we can try new things, and see their impact.
+
+For example, this year we added a Symbols server, which collects all the debug-files for all our OSes, so we can much easier/faster delve into crash dumps people send us.
+Before the Symbols server, there were a lot of manual actions required to open up a crash dump; now it is just pressing a single button and waiting for a bit of time.
+
+We also added much more caching to the wiki.
+As wiki pages do not change all that often, we can in fact cache them pretty aggressively, and only remove the cache entry when something changes.
+This means that if you go to our wiki now, Cloudflare serves almost everything from cache.
+It does check, in the background, if anything changed.
+And if so, it will serve the newest content.
+But as it is much more likely that it already has the content, it will serve it nearly instantly.
+For you this means a much faster website; for us it means a lower AWS bandwidth bill.
+To put it in a dollar amount, this method of caching saves us roughly 10 dollar per month.
```

Comment:
Most of the post talks about relative cost (x% of yearly donations).
Using an explicit number here is weird, because it names the "cost for a coffee-to-go in the west", without ever relating it to the fraction of a cent donated by the average player :)